core/src/main/java/org/elasticsearch/common/unit/ByteSizeValue.java

```diff
@@ -51,8 +65,23 @@ public ByteSizeValue(long bytes) {
     public ByteSizeValue(long size, ByteSizeUnit unit) {
         this.size = size;
         this.unit = unit;
+        if (size > Long.MAX_VALUE / unit.toBytes(1)) {
+            throw new IllegalArgumentException(
+                    "Values greater than " + Long.MAX_VALUE + " bytes are not supported: " + size + unit.getSuffix());
+        }
```

Comment:
should it also fail if size is < -1 or if size == -1 and unit != BYTES?

Reply:
Agreed, see https://github.com/elastic/elasticsearch/pull/27702/commits/ad16a5ce33b4b3b045f67aa5d5471a0ed68597d2